packages/components/src/entities/utils.tsx

```diff
@@ -0,0 +1,378 @@
+}
+
+/**
+ * Provides sample wizard URL for this application.
+ * @param targetSampleSet - Intended sample type of newly created samples.
+ * @param parent - Intended parent of derived samples. Format SCHEMA:QUERY:ID
+ * @param selectionKey
+ * @param currentProductId
+ * @param targetProductId
+ */
+export function getSampleWizardURL(
+    targetSampleSet?: string,
+    parent?: string,
+    selectionKey?: string,
+    currentProductId?: string,
+    targetProductId?: string
+): string | AppURL {
+    const params = {};
+
+    if (targetSampleSet) {
+        params['target'] = targetSampleSet;
+    }
+
+    if (parent) {
+        params['parent'] = parent;
+    }
+
+    if (selectionKey) params['selectionKey'] = selectionKey;
+
+    return createProductUrlFromParts(targetProductId, currentProductId, params, SAMPLES_KEY, 'new');
+}
+
+// TODO: Convert this into a component and utilize useServerContext() to fetch moduleContext for isELNEnabled() check
+export function getSampleDeleteMessage(canDelete: boolean, deleteInfoError: boolean): ReactNode {
+    let deleteMsg;
+    if (canDelete === undefined) {
+        deleteMsg = <LoadingSpinner msg="Loading delete confirmation data..." />;
+    } else if (!canDelete) {
+        deleteMsg = 'This sample cannot be deleted because ';
+        if (deleteInfoError) {
+            deleteMsg += 'there was a problem loading the delete confirmation data.';
+        } else {
+            deleteMsg += 'it has either derived sample, job, or assay data dependencies, ';
+            if (isELNEnabled()) {
+                deleteMsg += 'status that prevents deletion, or references in one or more active notebooks';
+            } else {
+                deleteMsg += 'or status that prevents deletion';
+            }
+            deleteMsg += '. Check the Lineage, Assays, and Jobs tabs for this sample to get more information.';
+        }
+    }
+    return deleteMsg;
+}
+
+export const getSampleTypeTemplateUrl = (
+    queryInfo: QueryInfo,
+    importAliases: Record<string, string>,
+    excludeColumns: string[] = ['flag', 'Ancestors'],
+    exportConfig: any = SAMPLE_EXPORT_CONFIG
+): string => {
+    const { schemaQuery } = queryInfo;
+    if (!schemaQuery) return undefined;
+
+    const extraColumns = SAMPLE_INSERT_EXTRA_COLUMNS.concat(Object.keys(importAliases || {})).filter(
+        col => excludeColumns.indexOf(col) == -1
+    );
+
+    return ActionURL.buildURL('query', 'ExportExcelTemplate', null, {
+        ...exportConfig,
+        schemaName: schemaQuery.getSchema(),
+        'query.queryName': schemaQuery.getQuery(),
+        headerType: 'DisplayFieldKey',
+        excludeColumn: excludeColumns
+            ? excludeColumns.concat(queryInfo.getFileColumnFieldKeys())
+            : queryInfo.getFileColumnFieldKeys(),
+        includeColumn: extraColumns,
+    });
+};
+
+export const getOriginalParentsFromLineage = async (
```

Comment:
This makes API calls to the server, it is not a util, it belongs in an API wrapper.

Reply:
I moved this to /actions.ts for now. As per chat conversation, I agree we'd like to move towards an API wrapper instead of the actions.ts file but I wasn't going to take that on for this PR. I did add a TODO at the top of the actions.ts file saying "don't add anything else here and instead create API wrapper for anything new"